src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -136,6 +136,9 @@ std::shared_ptr<ov::ICompiledModel> Plugin::compile_model(const std::shared_ptr<
 
     const auto& device_name = meta_device.device_name;
     const auto& device_config = meta_device.device_config;
+    if (auto pos = device_name.find("-") != std::string::npos) {
+        OPENVINO_THROW("Invalid device name '", device_name, "' for BATCH");
+    }
```

Comment:
Will be better if move this checking into `validate_batch_device_name`?

Reply:
I've moved it to `validate_batch_device_name`. :)